plugin/src/main/java/org/opensearch/ml/helper/MemoryEmbeddingHelper.java

```diff
@@ -0,0 +1,358 @@
+            // Create MLInput for text embedding with multiple documents
+            MLInput mlInput = MLInput
+                .builder()
+                .algorithm(embeddingModelType)
+                .inputDataset(TextDocsInputDataSet.builder().docs(texts).build())
+                .build();
+
+            // Create prediction request
+            MLPredictionTaskRequest predictionRequest = MLPredictionTaskRequest
+                .builder()
+                .modelId(embeddingModelId)
+                .mlInput(mlInput)
+                .build();
+
+            // Execute prediction
+            client.execute(MLPredictionTaskAction.INSTANCE, predictionRequest, ActionListener.wrap(response -> {
+                try {
+                    MLOutput mlOutput = response.getOutput();
+                    if (mlOutput instanceof ModelTensorOutput) {
+                        ModelTensorOutput tensorOutput = (ModelTensorOutput) mlOutput;
+                        List<Object> embeddings = new ArrayList<>();
+
+                        if (tensorOutput.getMlModelOutputs() != null) {
+                            for (ModelTensors modelTensors : tensorOutput.getMlModelOutputs()) {
+                                Object embedding = null;
+                                if (embeddingModelType == FunctionName.TEXT_EMBEDDING) {
+                                    embedding = extractDenseEmbeddingFromModelTensors(modelTensors);
+                                } else if (embeddingModelType == FunctionName.SPARSE_ENCODING) {
+                                    embedding = extractSparseEmbeddingFromModelTensors(modelTensors);
+                                }
+                                embeddings.add(embedding);
+                            }
+                        }
+
+                        listener.onResponse(embeddings);
+                    } else {
+                        log.error("Unexpected ML output type: {}", mlOutput.getClass().getName());
+                        listener.onFailure(new IllegalStateException("Unexpected ML output type: " + mlOutput.getClass().getName()));
+                    }
+                } catch (Exception e) {
+                    log.error("Failed to extract embeddings from ML output", e);
+                    listener.onFailure(new IllegalStateException("Failed to extract embeddings from ML output", e));
+                }
+            }, e -> {
+                log.error("Failed to generate embeddings", e);
+                listener.onFailure(e);
+            }));
+        }, listener::onFailure));
+    }
+
+    /**
+     * Generate embedding for a single text
+     * 
+     * @param text text to embed
+     * @param storageConfig memory storage configuration
+     * @param listener action listener for the result
+     */
+    public void generateEmbedding(String text, MemoryStorageConfig storageConfig, ActionListener<Object> listener) {
```

Comment:
See duplicate code with `generateEmbeddingsForMultipleTexts` , maybe we can merge?